### PR TITLE
Search by applicant, API only

### PR DESCRIPTION
**Pull request by luontola**
POC of #1352 and #873

### CODE REVIEW

File: src/clj/rems/standalone.clj [hunk 1]
```diff
@@ -39,14 +41,21 @@
...
 (defn start-app [& args]
   (doseq [component (-> args
                         (parse-opts cli-options)
                         mount/start-with-args
                         :started)]
     (log/info component "started"))
   (.addShutdownHook (Runtime/getRuntime) (Thread. stop-app))
-  (validate/validate))
+  (validate/validate)
+  (refresh-caches))
```
Comment (opqdonut) [inline src/clj/rems/standalone.clj]: excellent

File: src/clj/rems/application/search.clj [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+           [org.apache.lucene.search IndexSearcher ScoreDoc TopDocs SearcherManager SearcherFactory]
+           [org.apache.lucene.store Directory NIOFSDirectory]))
+
+(def ^:private analyzer (StandardAnalyzer.))
+
+(mount/defstate ^Directory search-index
+  :start (let [directory (NIOFSDirectory. (.toPath (io/file (:search-index-path env))))]
+           (with-open [writer (IndexWriter. directory (-> (IndexWriterConfig. analyzer)
+                                                          (.setOpenMode IndexWriterConfig$OpenMode/CREATE)))]
+             (.deleteAll writer))
```
Comment (opqdonut) [inline src/clj/rems/application/search.clj]: right so the indexes are cleared every restart. that's good.

File: src/clj/rems/application/search.clj [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+           [org.apache.lucene.document Document StringField Field$Store TextField]
+           [org.apache.lucene.index IndexWriter IndexWriterConfig IndexWriterConfig$OpenMode]
+           [org.apache.lucene.queryparser.flexible.standard StandardQueryParser]
+           [org.apache.lucene.search IndexSearcher ScoreDoc TopDocs SearcherManager SearcherFactory]
+           [org.apache.lucene.store Directory NIOFSDirectory]))
+
+(def ^:private analyzer (StandardAnalyzer.))
+
+(mount/defstate ^Directory search-index
+  :start (let [directory (NIOFSDirectory. (.toPath (io/file (:search-index-path env))))]
```
Comment (opqdonut) [inline src/clj/rems/application/search.clj]: should you guard against a a nil :search-index-path here?

File: src/clj/rems/application/search.clj [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+             (Long/parseLong id)))))
+
+(defn find-applications [^String query]
+  (refresh!) ; TODO: call from a background thread asynchronously?
+  (with-searcher
+   (fn [^IndexSearcher searcher]
+     (let [results (.search searcher
+                            (-> (StandardQueryParser. analyzer)
+                                (.parse query "applicant")) ; TODO: change defaultField to full text search
+                            10000)]
```
Comment (opqdonut) [inline src/clj/rems/application/search.clj]: nit: magic number

File: src/clj/rems/application/search.clj (hunk 3 above)
Comment (luontola) [reply src/clj/rems/application/search.clj]: There is a default for it in config-defaults.edn, so it should never be nil.

File: src/clj/rems/application/search.clj (hunk 4 above)
Comment (luontola) [reply src/clj/rems/application/search.clj]: Shall we use `Integer/MAX_VALUE` instead? Anyways we have no paging, but the UI shows all applications by default.

**Pull request merged**